src/components/highlight/highlight.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment, HTMLAttributes, FunctionComponent } from 'react';
+import { CommonProps } from '../common';
+import PropTypes from 'prop-types';
+
+export type EuiHighlightProps = HTMLAttributes<HTMLDivElement> &
```

Comment:
The `...rest` is spread onto a div element

```suggestion
export type EuiHighlightProps = HTMLAttributes<HTMLSpanElement> &
```